src/gluonnlp/model/bert.py

```diff
@@ -30,13 +30,212 @@
+        Initializer of the bias.
+    prefix : str or None, default None
+        See document of `Block`.
+    params : str or None, default None
+        See document of `Block`.
+
+    Inputs:
+      - **qkv** : Symbol or NDArray
+        Query / Key / Value vector. Shape (query_length, batch_size, C_in)
+      - **valid_len** : Symbol or NDArray or None, default None
+        Valid length of the query/key/value slots. Shape (batch_size, query_length)
+
+    Outputs:
+      - **context_vec** : Symbol or NDArray
+        Shape (query_length, batch_size, context_vec_dim)
+      - **att_weights** : Symbol or NDArray
+        Attention weights of multiple heads.
+        Shape (batch_size, num_heads, query_length, memory_length)
+    """
+    def __init__(self, units, num_heads, dropout=0.0, use_bias=True,
+                 weight_initializer=None, bias_initializer='zeros',
+                 prefix=None, params=None):
+        super().__init__(prefix=prefix, params=params)
+        self._num_heads = num_heads
+        self._use_bias = use_bias
+        self._dropout = dropout
+        self.units = units
+        with self.name_scope():
+            if self._use_bias:
+                self.query_bias = self.params.get('query_bias', shape=(self.units,),
+                                                  init=bias_initializer)
+                self.key_bias = self.params.get('key_bias', shape=(self.units,),
+                                                init=bias_initializer)
+                self.value_bias = self.params.get('value_bias', shape=(self.units,),
+                                                  init=bias_initializer)
+            weight_shape = (self.units, self.units)
+            self.query_weight = self.params.get('query_weight', shape=weight_shape,
+                                                init=weight_initializer,
+                                                allow_deferred_init=True)
+            self.key_weight = self.params.get('key_weight', shape=weight_shape,
+                                              init=weight_initializer,
+                                              allow_deferred_init=True)
+            self.value_weight = self.params.get('value_weight', shape=weight_shape,
+                                                init=weight_initializer,
+                                                allow_deferred_init=True)
+            self.dropout_layer = nn.Dropout(self._dropout)
+
+    def _collect_params_with_prefix(self, prefix=''):
+        # the registered parameter names in v0.8 are the following:
+        # prefix_proj_query.weight, prefix_proj_query.bias
+        # prefix_proj_value.weight, prefix_proj_value.bias
+        # prefix_proj_key.weight, prefix_proj_key.bias
+        # this is a temporary fix to keep backward compatibility, due to an issue in MXNet:
+        # https://github.com/apache/incubator-mxnet/issues/17220
+        if prefix:
+            prefix += '.'
```

Comment:
Doc says `prefix_` but u add `.`?

Reply:
this follows the original implementation: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/block.py#L409-L415
We only change the part after `proj`